### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by jworcestBAH**
Hi @kaitlin, we've now completed all the PRs to dev from sprint 8, so it seems like a good time to do a PR into master, if you'd like to look over the changes.

### CODE REVIEW

File: dataactbroker/app.py [hunk 1]
```diff
@@ -1,52 +1,106 @@
...
-    SessionTable.setup(app, runLocal)
-    app.run(debug=debugFlag,threaded=True,host="0.0.0.0",port= int(config["port"]))
+    try :
+        # Create application
+        config_path = "".join([os.path.dirname(os.path.abspath(
+            inspect.getfile(inspect.currentframe()))), "/config"])
+        app = Flask(__name__, instance_path=config_path)
+        config = getAppConfiguration(app)
+        if(config["local"]) :
+            # Up to 1 GB can be uploaded for local
```
Comment (kaitlin) [inline dataactbroker/app.py]: Why is there a size limit just for local?

File: dataactbroker/handlers/accountHandler.py [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+            username -- (string) the name of the  user
+            title  --   (string) the title of the  user
+            agency -- (string) the agency of the  user
+            userEmail -- (string) the email of the user
+            link  -- (string) the broker email link
+            """
+            threadedDatabase =  UserHandler()
+            try:
+                for user in threadedDatabase.getUsersByType("website_admin") :
+                    emailTemplate = {'[REG_NAME]': username, '[REG_TITEL]':title, '[REG_AGENCY]':agency,'[REG_EMAIL]' : userEmail,'[URL]':link}
```
Comment (kaitlin) [inline dataactbroker/handlers/accountHandler.py]: Should change REG_TITEL to REG_TITLE

File: dataactbroker/handlers/accountHandler.py [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+        try :
+            user = self.interfaces.userDb.getUserByEmail(requestFields.getValue("email"))
+        except ResponseException as e:
+            self.interfaces.userDb.addUnconfirmedEmail(email)
+        else:
+            if(not (user.user_status_id == self.interfaces.userDb.getUserStatusId("awaiting_confirmation") or user.user_status_id == self.interfaces.userDb.getUserStatusId("email_confirmed"))):
+                exc = ResponseException("User already registered", StatusCode.CLIENT_ERROR)
+                return JsonResponse.error(exc,exc.status)
+        emailToken = sesEmail.createToken(email,self.interfaces.userDb,"validate_email")
+        LoginSession.logout(session)
```
Comment (kaitlin) [inline dataactbroker/handlers/accountHandler.py]: why log the user out here?

File: dataactbroker/handlers/aws/session.py [hunk 4]
```diff
@@ -192,7 +279,7 @@ class SessionTable :
     KEY_NAME = "uid"
     DATA_FIELD = "data"
     DATE_FIELD = "expiration"
-    TIME_OUT_LIMIT = 1800
+    TIME_OUT_LIMIT = 604800
     TableConnection = ""
     isLocal = False
     localPort =  8000
```
Comment (kaitlin) [inline dataactbroker/handlers/aws/session.py]: isn't there a config option for local port that should be inserted here?

File: dataactbroker/handlers/userHandler.py [hunk 5]
```diff
@@ -14,26 +20,388 @@ class UserHandler(UserInterface):
...
+    def getPermssionList(self):
+        """ Gets the permission list
+
+        Returns:
+            list of PermissionType objects
+        """
+        queryResult = self.session.query(PermissionType).all()
+        return queryResult
+
+    def createUserWithPassword(self,email,password,bcrypt,admin=False):
```
Comment (kaitlin) [inline dataactbroker/handlers/userHandler.py]: where can this be triggered from? Just the command line?

**Comment by kaitlin** [comment]
Made a few comments, finished reviewing.

File: dataactbroker/handlers/accountHandler.py (hunk 2 above)
Comment (jworcestBAH) [inline dataactbroker/handlers/accountHandler.py]: Fixed

File: dataactbroker/handlers/aws/session.py (hunk 4 above)
Comment (jworcestBAH) [inline dataactbroker/handlers/aws/session.py]: Yes, this is a default value that is overwritten by the value in the config file.  I've renamed the variable to LOCAL_PORT to make it more consistent with other fields we're treating in the same way, and added a comment to clarify.  app.py reads from the config file and writes the values to the appropriate variables.

File: dataactbroker/handlers/userHandler.py (hunk 5 above)
Comment (jworcestBAH) [inline dataactbroker/handlers/userHandler.py]: This function is used by the unit tests and the configuration script to directly create initial users without going through the registration process.  I've added some more detail to the docstring for this function.

Comment (AileenBurgman) [inline dataactbroker/handlers/userHandler.py]: I will be out of the office starting  03/25/2016 and will not return until
04/04/2016.

I will return your message as soon as possible.  Thanks.

File: dataactbroker/app.py (hunk 1 above)
Comment (jworcestBAH) [inline dataactbroker/app.py]: Removed this requirement.

File: dataactbroker/handlers/accountHandler.py (hunk 3 above)
Comment (jworcestBAH) [inline dataactbroker/handlers/accountHandler.py]: I've removed this logout, it is not required.

**Comment by jworcestBAH** [comment]
Replacing this with a PR from staging to master.

**Pull request closed without merging**